client/state/selectors/get-post-type-all-posts-url.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getSiteSlug, isJetpackSite, isSingleUserSite } from 'state/sites/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+export const getPostTypeAllPostsUrl = ( state, postType ) => {
+	const siteId = getSelectedSiteId( state );
+	const siteSlug = getSiteSlug( state, siteId );
+	const isJetpack = isJetpackSite( state, siteId );
+	const isSingleUSer = isSingleUserSite( state, siteId );
+
+	const postTypeUrl = get( { page: 'pages', post: 'posts' }, postType, `types/${ postType }` );
+
+	if ( postType === 'post' && ! isJetpack && ! isSingleUSer ) {
+		return `/${ postTypeUrl }/my/${ siteSlug }`;
```

Comment:
Any reason why this is redirecting to all _my_ posts and not to all the posts from everyone? Since the button label is "All posts" I expected to see all the posts as we do in WP Admin.

Reply:
This is replicating the behaviour of the sidebar and of other selectors of the same type. The reason may as well be historical.